src/mlpack/methods/ann/init_rules/gaussian_init.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file gaussian_init.hpp
+ * @author Kris Singh
+ *
+ * Intialization rule for the neural networks. This simple initialization is
+ * performed by assigning a gaussian matrix with a given mean and variance
+ * to the weight matrix.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_INIT_RULES_GAUSSIAN_INIT_HPP
+#define MLPACK_METHODS_ANN_INIT_RULES_GAUSSIAN_INIT_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * This class is used to initialize weigth matrix with a gaussian.
+ */
+class GaussianInitialization
+{
+ public:
+  /**
+   * Initialize the gaussian initialization rule with mean=0 and
+   * variance = 1.
+   *
+   * @param mean The mean of the gaussian.
+   * @param variance The variance of the gaussian.
+   */
+  GaussianInitialization():
+  mean(0), variance(1)
+  {}
+
+  /**
+   * Initialize the gaussian with the given mean and variance.
+   *
+   * @param mean Mean of the gaussian
+   * @param variance Variance of the gaussian
+   */
+  GaussianInitialization(const double mean=0, const double variance=1):
+  mean(mean), variance(variance)
+  {}
+
+  /**
+   * Initialize the elements weight matrix using a Gaussian Distribution.
+   *
+   * @param W Weight matrix to initialize.
+   * @param rows Number of rows.
+   * @param cols Number of columns.
+   */
+  void Initialize(arma::mat& W,
+                  const size_t rows,
+                  const size_t cols,
+                  const size_t seed =21)
+  {
+    std::random_device rd;
```

Comment:
I think ``randu``is probably faster if zero mean and unit variance. What do you think?